packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -246,6 +249,23 @@ class _BottomSheetState extends State<BottomSheet> {
     return false;
   }
 
+  @override
+  void dispose() {
+    _effectiveAnimationController.dispose();
+    super.dispose();
+  }
+
+  @override
+  void didUpdateWidget(covariant BottomSheet oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.animationController != null) {
+      _effectiveAnimationController.dispose();
+      _effectiveAnimationController = widget.animationController!;
+    } else {
+      _effectiveAnimationController = BottomSheet.createAnimationController(this);
```

Comment:
If `widget.animationController` was null before and still is null, you do not want to create a new controller. You can just reuse the old one.